src/nfts/NftGallery.tsx

```diff
@@ -0,0 +1,132 @@
+import colors from 'src/styles/colors'
+import fontStyles from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+import variables from 'src/styles/variables'
+
+// This is used instead of a gap between the images because the gap is not supported in RN yet.
+const imageSize = (variables.width - Spacing.Regular16 * 3) / 2
+
+export default function NftGallery() {
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
+  const loading = useSelector(nftsLoadingSelector)
+  const error = useSelector(nftsErrorSelector)
+  const nfts = useSelector(nftsSelector)
+
+  useEffect(() => {
+    ValoraAnalytics.track(NftEvents.nft_gallery_screen_open)
+    dispatch(fetchNfts())
+  }, [])
+
+  return (
+    <SafeAreaView testID="NftGallery" style={styles.safeAreaContainer} edges={['top']}>
+      <DrawerTopBar middleElement={<Text>{t('nftGallery.title')}</Text>} />
+      {error ? (
+        <NftsLoadError testID="NftGallery/NftsLoadErrorScreen" />
+      ) : (
+        <FlatList
+          numColumns={2}
+          data={nfts}
+          refreshing={loading}
+          onRefresh={() => dispatch(fetchNfts())}
+          contentContainerStyle={styles.contentContainer}
+          refreshControl={
+            <RefreshControl
+              refreshing={loading}
+              colors={[colors.greenUI]}
+              onRefresh={() => dispatch(fetchNfts())}
+            />
+          }
+          renderItem={({ item, index }) => (
+            <Touchable
+              borderless={false}
+              onPress={() => navigate(Screens.NftsInfoCarousel, { nfts: [item] })}
+              style={[
+                styles.touchableIcon,
+                index % 2 === 0
+                  ? { marginRight: Spacing.Regular16 } &&
+                    index === nfts.length - 1 &&
+                    styles.lastImage
+                  : { marginLeft: Spacing.Regular16 },
+              ]}
+            >
+              <NftImage
+                nft={item}
+                testID="NftGallery/NftImage"
+                width={imageSize}
+                height={imageSize}
+                ErrorComponent={
+                  <View style={styles.errorView}>
+                    <ImageErrorIcon color="#C93717" />
+                  </View>
+                }
+                origin={NftOrigin.NftGallery}
+                borderRadius={Spacing.Smallest8}
+              />
+            </Touchable>
+          )}
+          keyExtractor={(item) => `${item.contractAddress}-${item.tokenId}`}
+          ListEmptyComponent={
+            loading ? null : (
+              <View style={styles.noNftsView}>
+                <Text style={styles.noNftsText}>{t('nftGallery.noNfts')}</Text>
+              </View>
+            )
+          }
+        />
+      )}
+    </SafeAreaView>
+  )
+}
+
+const styles = StyleSheet.create({
+  contentContainer: {
+    alignItems: 'center',
+    paddingBottom: '12%', // TODO: find a better way to do this
```

Comment:
why does this need to be a percentage? can we just use a Spacing constant?

Reply:
I tried the largest spacing constant and the last image was still covered by the bottom bar.

<p align="center">
<img src="https://github.com/valora-inc/wallet/assets/26950305/bb1a565a-8602-4b53-af62-e18a3edf0d35" width="50%" height="auto" />
</p>

Reply:
maybe we could apply a bottom padding on the images themselves? i'm not sure, but it seems like something we can tidy up later